src/command.ts

```diff
@@ -198,37 +223,37 @@ export default abstract class Command {
     }
 
     if (result && this.jsonEnabled()) {
-      CliUx.ux.styledJSON(this.toSuccessJson(result))
+      ux.styledJSON(this.toSuccessJson(result))
     }
 
     return result
   }
 
-  exit(code = 0): void {
+  public exit(code = 0): void {
     return Errors.exit(code)
   }
 
-  warn(input: string | Error): string | Error {
+  public warn(input: string | Error): string | Error {
     if (!this.jsonEnabled()) Errors.warn(input)
     return input
   }
 
-  error(input: string | Error, options: {code?: string; exit: false} & PrettyPrintableError): void
+  public error(input: string | Error, options: {code?: string; exit: false} & PrettyPrintableError): void
```

Comment:
Did you have something specific in mind? I don't think a conditional type would eliminate the need to overload this method. You still need to express that if `{exit: false} ` is provided, then return type is `void` otherwise it's `never`